### PR TITLE
#18 - jbrowse feedback

### PR DESCRIPTION
**Pull request by samrichca**
no changes to dms-jbrowse for this ticket.

### CODE REVIEW

File: components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx [hunk 1]
```diff
@@ -86,8 +87,17 @@ const JbrowseLaunchButton = () => {
           }: {
             node: JbrowseQueryNode;
           }) => checkJbrowseCompatibility({ file_access, file_type, index_file }),
-        ).length;
-        const canEnableJbrowse = jbrowseFileCount > 0 && jbrowseFileCount <= MAX_JBROWSE_FILES;
+        );
+        const jbrowseCompatibleFileCount = jbrowseCompatibleFiles.length;
+        // TEMP BAM LIMIT
+        const hasMultipleBams =
```
Comment (samrichca) [inline components/pages/explorer/Jbrowse/JbrowseLaunchButton.tsx:93]: caroline & jbrowse team are fixing the issue with displaying multiple BAM files.

File: components/pages/explorer/RepositoryContent.tsx [hunk 2]
```diff
@@ -145,13 +145,13 @@ const getTableConfigs = ({
...
         css: css`
-          &:nth-of-type(2n-1) {
+          &:nth-of-type(2n-1):not(.selected) {
             background-color: ${theme.colors.grey_1};
           }
         `,
         hoverBackground: theme.colors.grey_highlight,
         lineHeight: '1.5rem',
-        selectedBackground: 'pink',
+        selectedBackground: theme.colors.secondary_light,
```
Comment (samrichca) [inline components/pages/explorer/RepositoryContent.tsx:154]: change selected rows from pink to blue & override table row colour if that row is selected.

File: components/theme/icons/spinner.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ const Spinner = ({ fill, height, width }: IconProps) => {
       width={width}
       height={height}
       css={css`
-        height: 20px;
+        height: ${height}px;
```
Comment (samrichca) [inline components/theme/icons/spinner.tsx:39]: make spinner icon use the pre-existing height prop

File: pages/_document.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ const Document = () => {
         <link rel="preconnect" href="https://fonts.googleapis.com" />
         <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="anonymous" />
         <link
-          href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;600&amp;display=swap"
+          href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&amp;display=swap"
```
Comment (samrichca) [inline pages/_document.tsx:15]: lato doesn't have a 600 weight so i changed it to 700. bold fonts are bold now.

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx [hunk 5]
```diff
@@ -184,13 +184,14 @@ const JbrowseWrapper = () => {
         ),
       )
       .catch((error: Error) => handleError(error))
-      .finally(() => setJbrowseLoading(false));
+      .finally(() => setTimeout(() => setJbrowseLoading(false), 1000));
```
Comment (samrichca) [inline components/pages/explorer/Jbrowse/JbrowseWrapper.tsx:187]: add a loading overlay for one second. gonna workshop this with patrick once jbrowse is on QA.

File: components/pages/explorer/Jbrowse/JbrowseSelectedFilesTable.tsx [hunk 6]
```diff
@@ -150,32 +157,44 @@ const JbrowseSelectedFilesTable = () => {
           dismissible
           level="warning"
         >
-          Some files selected are not supported by JBrowse. Supported file types:{' '}
-          {jbrowseAllowedFileTypes.join(', ')}. Index files are required.
+          {/* TEMP BAM LIMIT */}
+          {hasMultipleBams ? (
```
Comment (samrichca) [inline components/pages/explorer/Jbrowse/JbrowseSelectedFilesTable.tsx:161]: we should have a way to show multiple errors or warnings, but that's post-MVP. for now i just need to limit BAM files to 1 file.

File: components/ErrorNotification.tsx [hunk 7]
```diff
@@ -49,6 +49,20 @@ const getIconDimensions = (size: ErrorSize) =>
     [ERROR_SIZES.SM]: { width: 18, height: 18 },
   }[size]);
 
+const getDismissIconDimensions = (size: ErrorSize) =>
```
Comment (samrichca) [inline components/ErrorNotification.tsx:52]: patrick asked to make the small error notification a bit smaller

File: components/SimpleTable.tsx [hunk 8]
```diff
@@ -33,37 +33,37 @@ export type TableRecord = Record<string, string | number>;
 const TableStyled = styled('table')`
   ${({ theme }: { theme: typeof defaultTheme }) => css`
     ${theme.typography.data};
+    border: 1px solid ${theme.colors.grey_3};
```
Comment (samrichca) [inline components/SimpleTable.tsx:36]: updated the styling of the table below the jbrowse visualization to more closely match the files table.

File: components/SimpleTable.tsx [hunk 9]
```diff
@@ -74,24 +74,32 @@ const SimpleTable = ({
...
-          ))}
-        </TrStyled>
-      </thead>
-      <tbody>
-        {tableData.map((row: TableRecord) => (
-          <TrStyled key={row.file_id}>
-            {Object.entries(row).map(([key, value]) => (
-              <TdStyled key={key}>{value}</TdStyled>
+    <div
+      css={css`
```
Comment (samrichca) [inline components/SimpleTable.tsx:78]: add horizontal scrolling if the selected files table is too wide.

File: components/pages/explorer/Jbrowse/JbrowseSelectedFilesTable.tsx [hunk 10]
```diff
@@ -150,32 +157,44 @@ const JbrowseSelectedFilesTable = () => {
...
+            'Only one BAM file can be visualized at a time.'
+          ) : (
+            <>
+              {incompatibleFilesCount} file{incompatibleFilesCount === 1 ? '' : 's'} selected{' '}
```
Comment (samrichca) [inline components/pages/explorer/Jbrowse/JbrowseSelectedFilesTable.tsx:165]: specifying the number of incompatible files in the warning message

File: components/pages/explorer/Jbrowse/JbrowseWrapper.tsx [hunk 11]
```diff
@@ -214,6 +215,25 @@ const JbrowseWrapper = () => {
               selectedFiles={jbrowseInput}
             />
             <JbrowseSelectedFilesTable />
+            {jbrowseLoading && (
```
Comment (justincorrigible) [inline components/pages/explorer/Jbrowse/JbrowseWrapper.tsx:218]: will this ever happen? looks as if this variable is ever true, then the spinner in 202 will be shown instead.

**Pull request merged**